MFVNeutralino/test/One2Two/datacard.py

```diff
@@ -180,18 +183,27 @@ def st(x, typ=float):
             l = h2l(x)
         else:
             raise ValueError('duh %r' % x)
-        fmt = '%.9g' if typ == float else '%s'
-        return ' '.join(fmt % typ(v) for v in l)
+        fmt = '%.9g' if typ == float else '%s'    
+       
+        if rescale_new_config :
+            if "h_bkg_dvv_rebin_" in x :
+                return ' '.join(fmt % typ(l[i]*scale_bkg[i]) for i in range(len(l)))
+            else :
+                return ' '.join(fmt % typ(l[i]*scale_signal[i]) for i in range(len(l)))
+        else :
+            return ' '.join(fmt % typ(v) for v in l)
 
     for year in years:
-        setattr(r, 'observed_%s'    % year, st('h_observed_%s'      % year))
-        setattr(r, 'bkg_rate_%s'    % year, st('h_bkg_dvv_rebin_%s' % year))
-        setattr(r, 'sig_rate_%s'    % year, st(r.hs_dvv_rebin[year]))
+        rescale_new_config=True
+        #setattr(r, 'observed_%s'    % year, st('h_observed_%s'      % year)) # FIXME note this is only removed for Peace's studies of signal/background eff!
+        setattr(r, 'observed_%s'    % year, st('h_bkg_dvv_rebin_%s' % year, rescale_new_config=rescale_new_config))
+        setattr(r, 'bkg_rate_%s'    % year, st('h_bkg_dvv_rebin_%s' % year, rescale_new_config=rescale_new_config))
+        setattr(r, 'sig_rate_%s'    % year, st(r.hs_dvv_rebin[year], rescale_new_config=rescale_new_config))
         setattr(r, 'sig_uncert_%s'  % year, st(r.hs_uncert[year]))
 
         ngen = r.ngens[year]
         setattr(r, 'ngen_%s'        % year, str(ngen))
-        setattr(r, 'sigmc_alpha_%s' % year, st([x / ngen for x in r.rates[year]]))
+        setattr(r, 'sigmc_alpha_%s' % year, st([x / ngen for x in r.rates[year]],rescale_new_config=rescale_new_config))
 
         bkg_uncert = get('h_bkg_uncert_%s' % year)
         setattr(r, 'bkg_uncert_%s' % year, bkg_uncert)
```

Comment:
Can we instead copy this file and name it something like `scaling_studies_datacard.py`? Since the modifications were only needed for this particular study, and the simpler that our standard datacard code is, the better, to avoid mistakes down the road.

Reply:
Sure. I will do just as you suggested.